pkg/osd_cluster_upgrader/upgrader.go

```diff
@@ -30,6 +30,7 @@ var (
 	steps                  UpgradeSteps
 	osdUpgradeStepOrdering = []upgradev1alpha1.UpgradeConditionType{
 		upgradev1alpha1.SendStartedNotification,
+		upgradev1alpha1.SendDelayedNotification,
```

Comment:
The change in this PR looks good.
But maybe we can merge the two SendNotification before upgrade together?
If it started, send start notification.
If it was not started after x minutes of schedule, send delay notification.

Reply:
And maybe we do the same after the upgrade.
We should either send failed or complete notification at the end?

Reply:
Thanks for pointing that out. I agree, that is a better way to approach this. As I started writing the code for handling the 'cancelled' notification it became clear that they should all be unified - I'll work on that refactor.

Reply:
So ultimately I've ended up keeping these two separate - when I merged them into a bigger function, it became a bit too large and confusing for my liking, with not much benefit - I kind-of preferred the separation of the start time and the delayed check, and if we ever need to remove the delay check later, we can just remove it from the list of ordered steps. I have, however, renamed the delay from "SendDelay" to "UpgradeDelayedCheck" so that it has a similar feel to other steps like the PreUpgradeHealthCheck.

I'm definitely still willing to revisit this if you like, and please do let me know if you'd like me to pursue the merge regardless - just letting you know where I fell on it after some further updates.

Regarding a merge of failed/complete.. 'failed' is a special case that needs handling outside of the steps, as I've discovered (which you'll see in the most recent commit)